### DISCUSSION TITLE
Naming a class Data

### DISCUSSION THREAD

**Discussion by apcode** [pyg-team/pytorch_geometric#7360] [General]
Who in their right mind named a class in torch_geometric "Data" ?

It's like calling a class "letters".  It says nothing of its purpose or what it holds.

Graph or even GraphData is surely better. Given the class contains node values and an edge list.

**Comment by rusty1s** [comment]
Thanks for creating this amazing discussion :) Always good to be reminded that open-source contributions are a beautiful thing.

> Who in their right mind named a class in torch_geometric "Data" ?

I did.

I think it exactly describes the purpose, because it means that this object holds data that can be then used as input into a neural network. This class is not solely related to graphs, as it supports other types of data as well, e.g., meshes, point clouds, etc. That's why we did go for a more general name here. Not really sure why you are upset about this.

**Comment by apcode** [comment]
Thanks for responding. Sarcasm aside.

People have scripts to download_data, process_data, load_data and on and on and on.

None of these are the same "data" as this class is named. The term is really too unspecific and only can cause name clashes and naming awkwardness. It is not "exact" in any way, as in precise. Data is not a purpose. 

You actually describe the purpose "input into a neural network".  So a better name (though still not great) might have been NetworkInput. That would at least specify its purpose and be clear as to why this "Data" is different to the "Data" you just downloaded, or some other "Data".

And yes, it is open source, and no, of course I am not upset.
